internal/server/receiver_handlers.go

```diff
@@ -94,7 +95,7 @@ func (s *ReceiverServer) handlePayload() func(w http.ResponseWriter, r *http.Req
 			}
 
 			for _, resource := range receiver.Spec.Resources {
-				if err := s.annotate(ctx, resource, receiver.Namespace); err != nil {
+				if err := s.annotateCrossNamespaceReferences(ctx, resource, receiver.Namespace); err != nil {
```

Comment:
Testing this pr, in the logs of the notification controller I see
```
{"level":"info","ts":"2023-03-13T19:54:31.675+0100","logger":"receiver-server","msg":"resource 'ImageRepository/*.default' annotated","reconciler kind":"Receiver","name":"webapp","namespace":"flux-system"}
```

which isn't super informative (It gives this log even when no resources with the labels were found).
I think we might need to pass the logger down to the `annotateCrossNamespaceReferences` to give better logs when each resource is annotated (Wdyt? cc @makkes )

Reply:
Agreed, redone logging a bit, pls check again 🙏🏻

Reply:
Any reason why you are loggin at debug level? The log doesn't show up with the default log option
```
s.logger.V(1).Info(fmt.Sprintf("resource '%s/%s.%s' annotated",
					resource.Kind, resource.Name, namespace))
```

Secondly,  the log no longer has the receiver(name, ns) details associated with it. I am thinking we might want to pass in the logger created with these details into the function rather than use the one in the struct. It would help to differentiate logs for one receiver from another.

Reply:
Yes on purpose because this could target many resources and could be logged multiple times. There is a new log before individual resources. Let me know if it should be changed to info.

Reply:
>  I am thinking we might want to pass in the logger created with these details into the function rather than use the one in the struct. It would help to differentiate logs for one receiver from another.

yes makes sense, it's changed now